src/overset/UpdateOversetFringeAlgorithmDriver.C

```diff
@@ -36,13 +37,28 @@ void UpdateOversetFringeAlgorithmDriver::execute()
 {
   auto* oversetManager = realm_.oversetManager_;
   if (oversetManager->oversetGhosting_ != nullptr) {
+#ifndef KOKKOS_ENABLE_CUDA
     std::vector<const stk::mesh::FieldBase*> fVec(fields_.size());
     for (size_t i=0; i < fields_.size(); ++i)
       fVec[i] = fields_[i].field_;
     stk::mesh::communicate_field_data(
       *oversetManager->oversetGhosting_, fVec);
+#else
+    throw std::runtime_error("Overset ghosting not supported for GPU builds");
+#endif
+  }
+
+  const auto& fieldMgr = realm_.mesh_info().ngp_field_manager();
+  for (auto& finfo: fields_) {
+    auto& ngpField = fieldMgr.get_field<double>(finfo.field_->mesh_meta_data_ordinal());
+    ngpField.sync_to_host();
   }
   oversetManager->overset_update_fields(fields_);
+  for (auto& finfo: fields_) {
+    auto& ngpField = fieldMgr.get_field<double>(finfo.field_->mesh_meta_data_ordinal());
```

Comment:
Why not do these syncs and modify calls inside overset_update_fields?

Reply:
Because TIOGA classes don't know about Realm or MeshInfo.

Reply:
Note that the modify_on_* and sync_to_* calls are available on stk::mesh::FieldBase now, so you don't need an ngpField to call those. I guess if the TIOGA scope has no knowledge that there could be a GPU device, then it doesn't make sense for that scope to mark fields as modified nor to sync them.

Reply:
Oh! I didn't realize that.. that would make this code a lot concise.

Reply:
Thanks @alanw0. With the ability to use `modify/sync` methods on `FieldBase`, I was able to move the synchronization calls into the Tioga-STK interface class. This should address your _guiding convention_ comment in `Realm.C` also.